dbms/src/Common/tests/gtest_spiller.cpp

```diff
@@ -0,0 +1,506 @@
+            all_blocks[partition_id].insert(all_blocks[partition_id].end(), blocks.begin(), blocks.end());
+            spiller.spillBlocks(blocks, partition_id);
+        }
+    }
+    spiller.finishSpill();
+    for (size_t partition_id = 0; partition_id < partition_num; partition_id++)
+    {
+        size_t max_restore_streams = 2 + partition_id * 10;
+        auto restore_block_streams = spiller.restoreBlocks(partition_id, max_restore_streams);
+        size_t expected_streams = std::min(max_restore_streams, spill_num);
+        GTEST_ASSERT_EQ(restore_block_streams.size(), expected_streams);
+        Blocks all_restored_blocks;
+        for (const auto & block_stream : restore_block_streams)
+        {
+            for (Block block = block_stream->read(); block; block = block_stream->read())
+            {
+                all_restored_blocks.push_back(block);
+            }
+        }
+        GTEST_ASSERT_EQ(all_restored_blocks.size(), all_blocks[partition_id].size());
+        for (size_t i = 0; i < all_restored_blocks.size(); i++)
+        {
+            blockEqual(all_blocks[partition_id][i], all_restored_blocks[i]);
+        }
+    }
+}
+catch (Exception & e)
+{
+    GTEST_FAIL();
+}
+
+TEST_F(SpillerTest, SpillAndRestoreOrderedBlocks)
+try
+{
+    Spiller spiller("test", true, 2, spiller_test_tmp_dir, spiller_test_header, logger);
+    size_t partition_num = 2;
+    size_t spill_num = 5;
+    std::vector<Blocks> all_blocks(partition_num);
+    for (size_t partition_id = 0; partition_id < partition_num; partition_id++)
+    {
+        for (size_t spill_time = 0; spill_time < spill_num; spill_time++)
+        {
+            auto blocks = generateSortedBlocks(3);
+            all_blocks[partition_id].insert(all_blocks[partition_id].end(), blocks.begin(), blocks.end());
+            spiller.spillBlocks(blocks, partition_id);
+        }
+    }
+    spiller.finishSpill();
+    for (size_t partition_id = 0; partition_id < partition_num; partition_id++)
+    {
+        size_t max_restore_streams = 2 + partition_id * 10;
+        auto restore_block_streams = spiller.restoreBlocks(partition_id, max_restore_streams);
+        /// for sorted spill, the restored stream num is always equal to the spill time
+        size_t expected_streams = spill_num;
+        GTEST_ASSERT_EQ(restore_block_streams.size(), expected_streams);
+        Blocks all_restored_blocks;
+        for (const auto & block_stream : restore_block_streams)
+        {
+            for (Block block = block_stream->read(); block; block = block_stream->read())
+            {
+                all_restored_blocks.push_back(block);
+            }
+        }
+        GTEST_ASSERT_EQ(all_restored_blocks.size(), all_blocks[partition_id].size());
+        for (size_t i = 0; i < all_restored_blocks.size(); i++)
+        {
+            blockEqual(all_blocks[partition_id][i], all_restored_blocks[i]);
+        }
+    }
+}
+catch (Exception & e)
+{
+    GTEST_FAIL();
+}
```

Comment:
Or use the `CATCH` in `TiFlashTestBasic`?

Reply:
Done